Src/ILGPU.Algorithms/Runtime/Cuda/CuFFT.cs

```diff
@@ -0,0 +1,270 @@
+            API.Estimate1D(
+                nx,
+                type,
+                batch,
+                workSize);
+
+        /// <summary>
+        /// Estimates the work area for a 2D plan.
+        /// </summary>
+        /// <param name="nx">The transform size in the x dimension.</param>
+        /// <param name="ny">The transform size in the y dimension.</param>
+        /// <param name="type">The transform type.</param>
+        /// <param name="workSize">Filled in with the estimated size in bytes.</param>
+        /// <returns>The error code.</returns>
+        public CuFFTResult Estimate2D(
+            int nx,
+            int ny,
+            CuFFTType type,
+            UIntPtr[] workSize) =>
+            API.Estimate2D(
+                nx,
+                ny,
+                type,
+                workSize);
+
+        /// <summary>
+        /// Estimates the work area for a 3D plan.
+        /// </summary>
+        /// <param name="nx">The transform size in the x dimension.</param>
+        /// <param name="ny">The transform size in the y dimension.</param>
+        /// <param name="nz">The transform size in the z dimension.</param>
+        /// <param name="type">The transform type.</param>
+        /// <param name="workSize">Filled in with the estimated size in bytes.</param>
+        /// <returns>The error code.</returns>
+        public CuFFTResult Estimate3D(
+            int nx,
+            int ny,
+            int nz,
+            CuFFTType type,
+            UIntPtr[] workSize) =>
+            API.Estimate3D(
+                nx,
+                ny,
+                nz,
+                type,
+                workSize);
+
+        /// <summary>
+        /// Estimates the work area for a custom plan.
+        /// </summary>
+        /// <param name="rank">The transform.</param>
+        /// <param name="n">The transform dimensions.</param>
+        /// <param name="inembed">The storage dimensions of the input data.</param>
+        /// <param name="istride">The stride of the input data.</param>
+        /// <param name="idist">The distance of the input data.</param>
+        /// <param name="onembed">The storage dimensions of the output data.</param>
+        /// <param name="ostride">The stride of the output data.</param>
+        /// <param name="odist">The distance of the output data.</param>
+        /// <param name="type">The transform type.</param>
+        /// <param name="batch">The number of transforms.</param>
+        /// <param name="workSize">Filled in with the estimated size in bytes.</param>
+        /// <returns>The error code.</returns>
+        public CuFFTResult EstimateMany(
+            int rank,
+            int[] n,
+            int[] inembed,
+            int istride,
+            int idist,
+            int[] onembed,
+            int ostride,
+            int odist,
+            CuFFTType type,
+            int batch,
+            UIntPtr[] workSize) =>
```

Comment:
Question: should this also be Span types?

Reply:
Done.